pkg/sql/relocate_range.go

```diff
@@ -46,42 +46,33 @@ type relocateResults struct {
 	err       error
 }
 
-// relocateRequest is an internal data structure that describes a relocation.
-type relocateRequest struct {
-	rangeID         roachpb.RangeID
-	subjectReplicas tree.RelocateSubject
-	toStoreDesc     *roachpb.StoreDescriptor
-	fromStoreDesc   *roachpb.StoreDescriptor
-}
+func (n *relocateRange) startExec(params runParams) (err error) {
 
-func (n *relocateRange) startExec(params runParams) error {
-	toStoreID, err := paramparse.DatumAsInt(params.ctx, params.EvalContext(), "TO", n.toStoreID)
-	if err != nil {
-		return err
-	}
-	var fromStoreID int64
-	if n.subjectReplicas != tree.RelocateLease {
-		// The from expression is NULL if the target is LEASE.
-		fromStoreID, err = paramparse.DatumAsInt(params.ctx, params.EvalContext(), "FROM", n.fromStoreID)
+	typedExprToReplicationTarget := func(typedExpr tree.TypedExpr, name string) (target roachpb.ReplicationTarget, _ error) {
```

Comment:
The work in this function was being done 1x or 2x depending on `n.subjectReplicas`, but it was not obvious because it was copied and pasted across multiple `.subjectReplicas != tree.RelocateLease` conditionals.